client/src/app/project-registration/services/autofill-project.service.ts

```diff
@@ -0,0 +1,53 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import {map} from 'rxjs/operators';
+import {Observable, throwError} from 'rxjs';
+import {EuropePMCHttpSearchResponse, Identifier} from '../models/europepmcsearch';
+import {AutofillProject} from '../models/autofill-project';
+
+@Injectable()
+export class AutofillProjectService {
+  API_URL = 'https://www.ebi.ac.uk/europepmc/webservices/rest/search';
+  DOI_BASE_URL = 'https://doi.org/';
+
+  constructor(private http: HttpClient) {}
+
+  getProjectDetails(searchUsing: Identifier, searchString: string): Observable<AutofillProject> {
+    return this.queryEuropePMC(searchUsing, searchString).pipe(
+      map(response => {
+        const result = response.resultList.result[0];
+        if (result) {
+          const projectDetails: AutofillProject = {
+            title: result.title,
+            description: result.abstractText.replace(/(<([^>]+)>)/gi, ''),
+            doi: result.doi,
+            authors: result.authorString.replace('.', '').split(','),
+            pmid: result.pmid ? Number(result.pmid) : null,
+            url: this.DOI_BASE_URL + result.doi,
+            funders: result.grantsList && result.grantsList.grant ?
+              result.grantsList.grant.map(grant => ({grant_id: grant.grantId, organization: grant.agency})) : [],
+            contributors: result.authorList && result.authorList.author && result.authorList.author.length ?
+              result.authorList.author.map(author => ({
+                name: author.firstName + ',,' + author.lastName,
```

Comment:
2 commas?

Reply:
yes, that's the expected format of the name by contact name component: https://github.com/ebi-ait/ingest-ui/blob/aac7c7c65c46165f9007027fef81e07b52afc365/client/src/app/project-registration/contact-name-field/contact-name-field.component.ts#L31 and https://github.com/ebi-ait/ingest-ui/blob/aac7c7c65c46165f9007027fef81e07b52afc365/client/src/app/project-registration/contact-field-group/contact-field-group.component.ts#L60


I understand that's view specific logic so I've moved this from the service to the component